modules/command-engine/core/src/main/scala/surge/internal/core/SurgePartitionRouterImpl.scala

```diff
@@ -2,48 +2,108 @@
 import surge.health.HealthSignalBusTrait
-import surge.internal.SurgeModel
-import surge.internal.akka.kafka.KafkaConsumerPartitionAssignmentTracker
+import surge.internal.akka.kafka.{ KafkaConsumerPartitionAssignmentTracker, KafkaShardingClassicMessageExtractor }
 import surge.internal.config.TimeoutConfig
-import surge.internal.persistence.RoutableMessage
-import surge.kafka.streams.{ HealthCheck, HealthCheckStatus, HealthyActor, HealthyComponent }
+import surge.internal.persistence.PersistentActor
+import surge.internal.tracing.RoutableMessage
+import surge.internal.{ persistence, SurgeModel }
+import surge.kafka.streams._
 import surge.kafka.{ KafkaPartitionShardRouterActor, PersistentActorRegionCreator }
 
-import scala.concurrent.{ ExecutionContext, Future }
+import java.util.regex.Pattern
+import scala.concurrent.duration.DurationInt
+import scala.concurrent.{ Await, ExecutionContext, Future }
 import scala.languageFeature.postfixOps
 
 private[surge] final class SurgePartitionRouterImpl(
     config: Config,
     system: ActorSystem,
     partitionTracker: KafkaConsumerPartitionAssignmentTracker,
     businessLogic: SurgeModel[_, _, _, _],
+    aggregateKafkaStreamsImpl: AggregateStateStoreKafkaStreams[JsValue],
     regionCreator: PersistentActorRegionCreator[String],
-    signalBus: HealthSignalBusTrait)
+    signalBus: HealthSignalBusTrait,
+    isAkkaClusterEnabled: Boolean)
     extends SurgePartitionRouter
     with HealthyComponent
     with Controllable {
   implicit val executionContext: ExecutionContext = system.dispatcher
   private val log = LoggerFactory.getLogger(getClass)
 
-  private val shardRouterProps = KafkaPartitionShardRouterActor.props(
-    config,
-    partitionTracker,
-    businessLogic.partitioner,
-    businessLogic.kafka.stateTopic,
-    regionCreator,
-    RoutableMessage.extractEntityId)(businessLogic.tracer)
+  override val actorRegion: ActorRef = {
+    if (isAkkaClusterEnabled) {
+      createAkkaCluster()
+    } else {
+      createShardRouter()
+    }
+  }
+
+  log.info(s"Shard region: $actorRegion")
+
+  private def createShardRouter(): ActorRef = {
+    val shardRouterProps = KafkaPartitionShardRouterActor.props(
+      config,
+      partitionTracker,
+      businessLogic.partitioner,
+      businessLogic.kafka.stateTopic,
+      regionCreator,
+      RoutableMessage.extractEntityId)(businessLogic.tracer)
 
-  private val routerActorName = s"${businessLogic.aggregateName}RouterActor"
+    val routerActorName = s"${businessLogic.aggregateName}RouterActor"
+    val shardRouter = system.actorOf(shardRouterProps, name = routerActorName)
+    shardRouter
+  }
+
+  private def createAkkaCluster(): ActorRef = {
+    val groupId = businessLogic.kafka.streamsApplicationId
+    val consumerSettings = ConsumerSettings(system, new StringDeserializer, new ByteArrayDeserializer)
+      .withBootstrapServers(config.getString("kafka.brokers"))
```

Comment:
We can't leverage the KafkaSecurityConfiguration in this place. 
I've analyzed the [codebase](**url**) and it seems like we can't configure it manually. 
Possibly, we can pass security configs as described in this [docs](https://doc.akka.io/docs/alpakka-kafka/current/production.html)

Reply:
You should be able to pass in just a raw properties map via `.withProperties` - something like:
```
val securityRelatedProps = new Properties()
val securityHelper = new KafkaSecurityConfigurationImpl(config)
securityHelper.configureSecurityProperties(securityRelatedProps)

consumerSettings
  .withProperties(securityRelatedProps.asScala.toMap)
```

Reply:
Missed that, thank you! I've updated the kafka consumer settings